From 6336f877aa4be068fca24553b36b120578cdc6ac Mon Sep 17 00:00:00 2001
From: lambdadroid <lambdadroid@gmail.com>
Date: Thu, 7 Sep 2017 12:17:48 +0200
Subject: [PATCH 10/22] ME176C: pinctrl: baytrail: Remove warnings caused by
 hardware config

These warnings are printed whenever the screen is suspended. Since
there seems to be no way to fix the hardware config directly, just
remove them for now to make the kernel log more readable.
---
 drivers/pinctrl/intel/pinctrl-baytrail.c | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/drivers/pinctrl/intel/pinctrl-baytrail.c b/drivers/pinctrl/intel/pinctrl-baytrail.c
index 0f3a02495aeb..ff6891a913b5 100644
--- a/drivers/pinctrl/intel/pinctrl-baytrail.c
+++ b/drivers/pinctrl/intel/pinctrl-baytrail.c
@@ -1014,7 +1014,6 @@ static int byt_gpio_set_direction(struct pinctrl_dev *pctl_dev,
 {
 	struct byt_gpio *vg = pinctrl_dev_get_drvdata(pctl_dev);
 	void __iomem *val_reg = byt_gpio_reg(vg, offset, BYT_VAL_REG);
-	void __iomem *conf_reg = byt_gpio_reg(vg, offset, BYT_CONF0_REG);
 	unsigned long flags;
 	u32 value;
 
@@ -1024,15 +1023,6 @@ static int byt_gpio_set_direction(struct pinctrl_dev *pctl_dev,
 	value &= ~BYT_DIR_MASK;
 	if (input)
 		value |= BYT_OUTPUT_EN;
-	else
-		/*
-		 * Before making any direction modifications, do a check if gpio
-		 * is set for direct IRQ.  On baytrail, setting GPIO to output
-		 * does not make sense, so let's at least warn the caller before
-		 * they shoot themselves in the foot.
-		 */
-		WARN(readl(conf_reg) & BYT_DIRECT_IRQ_EN,
-		     "Potential Error: Setting GPIO with direct_irq_en to output");
 	writel(value, val_reg);
 
 	raw_spin_unlock_irqrestore(&vg->lock, flags);
@@ -1571,9 +1561,6 @@ static int byt_irq_type(struct irq_data *d, unsigned int type)
 	raw_spin_lock_irqsave(&vg->lock, flags);
 	value = readl(reg);
 
-	WARN(value & BYT_DIRECT_IRQ_EN,
-	     "Bad pad config for io mode, force direct_irq_en bit clearing");
-
 	/* For level trigges the BYT_TRIG_POS and BYT_TRIG_NEG bits
 	 * are used to indicate high and low level triggering
 	 */
-- 
2.16.2

